File: app/src/main/java/mx/yellowme/app/simple/UmbrellaFragment.java
package mx.yellowme.app.simple;

import android.util.Log;
import android.view.View;

import mx.yellowme.app.R;
import mx.yellowme.waterproof.mvp.simple.*;
import mx.yellowme.waterproof.mvp.simple.SimplePresenter;

/**
 * Created by migdonio on 8/1/17.
 */

public class UmbrellaFragment
        extends SimpleWaterproofFragment<Umbrella, UmbrellaViewHolder> {

    @Override
    public SimplePresenter<Umbrella> getPresenter() {
        return new UmbrellaPresenter(this);
    }

    @Override
    protected void setupAdapter() {
        mAdapter = new UmbrellaAdapter(this, mViewHolder);
    }

    @Override
    protected String getEmptyListMessage() {
        return "No hay sombrilla";
    }

    @Override
    public UmbrellaViewHolder getConcreteItemViewHolder(View root) {
        return new UmbrellaViewHolder(getContext(), root);
    }

    @Override
    public int getMainLayoutResource() {
        return R.layout.fragment_umbrella;
    }

}

File: app/src/main/java/mx/yellowme/app/simple/UmbrellaFragment.kt
package mx.yellowme.app.simple

import android.view.View

import mx.yellowme.app.R
import mx.yellowme.waterproof.mvp.simple.*
import mx.yellowme.waterproof.mvp.simple.SimplePresenter

/**
 * Created by migdonio on 8/1/17.
 */

class UmbrellaFragment : SimpleWaterproofFragment<Umbrella, UmbrellaViewHolder>() {

    override val presenter: SimplePresenter<Umbrella>
        get() = UmbrellaPresenter(this)

    override val emptyListMessage: String?
        get() = "No hay sombrilla"

    override val mainLayoutResource: Int
        get() = R.layout.fragment_umbrella

    override fun setupAdapter() {
        mAdapter = UmbrellaAdapter(this, mViewHolder)
    }

    override fun getConcreteItemViewHolder(root: View): UmbrellaViewHolder {
        return UmbrellaViewHolder(context, root)
    }

}
